src/w3d/renderer/motchan.cpp

```diff
@@ -432,14 +432,15 @@ AdaptiveDeltaMotionChannelClass::AdaptiveDeltaMotionChannelClass() :
     m_type(0),
     m_vectorLen(0),
     m_numFrames(0),
-    m_scale(0),
+    m_scale(0.0f),
     m_data(nullptr),
     m_cacheFrame(0),
     m_cacheData(nullptr)
 {
     if (!g_tableValid) {
+        const double rad = DEG_TO_RAD(90.0);
         for (int i = 0; i < 240; ++i) {
-            g_filterTable[i + 16] = 1.0 - GameMath::Sin(i / 240.0 * DEG_TO_RAD(90.0));
+            g_filterTable[i + 16] = 1.0f - GameMath::Sin(((float)i / 240.0f) * rad);
```

Comment:
there isn't any point adding the cast and just clutters the code:

Reply:
Is i / 240f treated correctly as float division though? It looks as if it could result in integer math.

Reply:
yes. It always promotes. 
https://en.cppreference.com/w/cpp/language/operator_arithmetic#Conversions

> Otherwise, if either operand is float, the other operand is converted to float

Reply:
ok